manifests/naemon_monitor.pp

```diff
@@ -137,6 +137,18 @@
     notify  => Class['sunet::systemd_reload'],
   }
 
+  # Make sure that a user with the name "naemon" is created in the database, with the same password as in Hiera.
+  # E.g `CREATE USER naemon IDENTIFIED BY '<something secret>';`
+  $check_mariadb_password = lookup('check_mariadb_password', Variant[String, Undef], undef, undef)
```

Comment:
Do you want to set a default, like NOT_SET_IN_HIERA and check for that, it seems a bit fragile to rely on undef (or is it Undef like in the Variant definition?)

Reply:
Can agree what it's easier to understand the logic if it's only Strings involved. Updated.